blog/2024-06-07-tagging-keys.mdx

```diff
@@ -0,0 +1,114 @@
+---
+slug: organize-your-translation-keys-with-tags
+title: Organize Your Translation Keys Automatically with Tags!
+description: A guide to the new Tolgee CLI feature that simplifies key organization with automatic extraction and tags.
+authors: [sgranat]
+tags: [tolgee, cli, tags, branching, tutorial]
+---
+
+Do you ever struggle with deprecated localization keys? Wondering which keys are active in production or still in draft pull requests? 😵‍💫
+
+Keeping localization keys organized in a large project can be a headache, especially with many contributors. But it’s crucial for a smooth localization process.
+
+<!-- truncate-->
+
+## The Challenge 🐭
+
+One of my daily tasks at Tolgee Platform is tracking new localization keys. I need to identify keys linked to new features since they’re in "draft" mode. Until the feature is released, everything is fluid—strings change, keys get added or removed, and sometimes entire features get scrapped. It’s easy to end up with unused keys that are forgotten and never removed.
+
+### The Bigger Picture ✨
+
+As we add new features, we inevitably have multiple versions (branches) of our application: the production version and feature branches. It’s essential to know which keys belong to which branch since production keys require careful handling, keys in development can be modified or removed as needed.
+
+## The key lifecycle ♻️
+
+Each key goes through a lifecycle:
+
+1. Added to the code (`draft`)
+2. Released to the public (`production`)
+3. No longer needed (`deprecated`)
+
+Different companies have different needs, but here’s the system we use at Tolgee:
+
+### 1. `draft` 🌱
+
+With the Tolgee SDK, you can set a default tag for new keys:
+
+```ts
+Tolgee().init({
+  tagNewKeys: ['draft'],
+})
+```
+
+When adding a new key through the Tolgee In-context tool, it appears like this:
+
+import { ScreenshotWrapper } from '../platform/shared/_ScreenshotWrapper';
+
+<ScreenshotWrapper src="/img/blog/tagging-keys/in-context-with-draft.png" alt="In-context with draft" />
+
+To identify different features, we store the branch name in a file, which is imported into our web application and used in the tag:
+
+```ts
+import { branchName } from './branch.json'
+
+Tolgee().init({
+  tagNewKeys: [`draft: ${branchName}`],
+})
+```
+
+`branch.json`:
+```json
+{
+  "branchName": "current-feature-branch"
+}
+```
+
+The `branch.json` file is excluded from git and updated with `post-commit` and `post-checkout` git hooks, so it's always up-to-date.
+
+
+> You can check our [github repository](https://github.com/tolgee/tolgee-platform) for full implementation.
```

Comment:
I would maybe add the whole setup. They won't find it in the repo. It's not that easy, it requires husky, it's in the webapp folder, there is lot of other stuff happening in the repo.